bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/communication/SonnenJSONCommunication.java

```diff
@@ -49,21 +51,57 @@ public SonnenJSONCommunication() {
      *
      * @return an empty string if no error occurred, the error message otherwise.
      */
-    public String refreshBatteryConnection() {
+    public String refreshBatteryConnection(boolean newAPI) {
```

Comment:
Consider providing two different methods, one for old API and one for new API. This combined method seems to branch into two different versions anyway, so method does two different things for no apparent reason.

Reply:
Of course I could make two different methodes for it, but it does not really make a difference in my opinion.

Reply:
Your proposed method has a cyclomatic complexity of 2, and it's also limited to two API versions. Splitting into two methods would make the code simpler and cleaner. See for example https://alexkondov.com/should-you-pass-boolean-to-functions/

Reply:
ok, will change it with the next commit.